### PR TITLE
Use the same method to build a random string for the code verifier an…

### PR DESCRIPTION
**Pull request by robinsoc**
…d state identifiers

This is a part of [story #14018] (https://tuleap.net/plugins/tracker/?aid=14018) have a central management of users and groups using oauth

No functional change expected.

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java [hunk 1]
```diff
@@ -154,7 +154,7 @@ protected String getPostLogOutUrl(StaplerRequest req, Authentication auth) {
     public HttpResponse doCommenceLogin(StaplerRequest request) throws UnsupportedEncodingException, NoSuchAlgorithmException {
         this.injectInstances();
 
-        final String state = this.buildStateRandomString();
```
Comment (Erwyn) [inline src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java:157]: Can we remove the corresponding method from the class ?

File: src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java [hunk 2]
```diff
@@ -154,7 +154,7 @@ protected String getPostLogOutUrl(StaplerRequest req, Authentication auth) {
...
+        final String state = this.pluginHelper.buildRandomBase64EncodedURLSafeString(30);
```
Comment (Erwyn) [inline src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java:157]: Why **30** ? If there is a special reason about this, can we make a constant out of this value ?

Comment (robinsoc) [reply src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java]: The 30 value can be explained here: 
https://tools.ietf.org/html/rfc6749#section-10.10

Comment (LeSuisse) [reply src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java]: This section is about the authorization server, you are not the authorization server. A 32 bytes octets in both cases is fine and recommended.

Furthermore, the maximum 160 bits of entropy is mostly an out of date recommendation that is not present anymore in recent security RFC on OAuth 2.0 (see RFC 6819 section 5.1.4.2.2). It's very unlikely that systems capable of handling a full OAuth 2.0 process (so with the capability to communicate over TLS) will struggle to deal with the storage of a string with a few more characters than needed anyway.

**Pull request merged**